src/operator/src/insert.rs

```diff
@@ -231,16 +260,37 @@ impl Inserter {
     async fn group_requests_by_peer(
         &self,
         requests: RegionInsertRequests,
-    ) -> Result<HashMap<Peer, RegionInsertRequests>> {
-        let mut inserts: HashMap<Peer, RegionInsertRequests> = HashMap::new();
+    ) -> Result<HashMap<PeerTyped, RegionInsertRequests>> {
+        let mut inserts: HashMap<PeerTyped, RegionInsertRequests> = HashMap::new();
 
         for req in requests.requests {
```

Comment:
Can we group requests in `table_id` for flow-grouping and in `region_id` for datanode-grouping first to reduce calling to `nodes` and `find_region_leader`?